.github/workflows/push.yml

```diff
@@ -8,56 +7,89 @@ on:
       - '**'
 
 jobs:
-  lint:
+  pre-checks:
     runs-on: ubuntu-latest
     steps:
       # Install golang
       - uses: actions/setup-go@v2
         with:
-          go-version: '^1.13.1'
+          go-version: 1.14
 
-      # Setup gopath
-      - name: Setting up GOPATH 
+      # Checkout to the latest commit
+      # On specific directory/path
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: gofmt check
         run: |
-          echo "GOPATH=${GITHUB_WORKSPACE}/go" >> $GITHUB_ENV
-        
+          if [ "$(gofmt -s -l . | wc -l)" -ne 0 ]
+          then
+           echo "The following files were found to be not go formatted:"
+           gofmt -s -l .
+           echo "Please run 'make fmt' to go format the above files."
+           exit 1
+          fi
+
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v1
+
+      - name: unused-package check
+        run: |
+          make unused-package-check
+
+  image-build:
+    runs-on: ubuntu-latest
+    needs: pre-checks
```

Comment:
we can make image-build and tests parallel but not pre-check because if code has some syntax or lint problem then the image will build unnecessary.   We did the same thing in litmus portal